tests/hardhat/XVS/XVSVault.ts

```diff
@@ -0,0 +1,115 @@
+import { loadFixture, mine } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { BigNumber, Wallet } from "ethers";
+import { ethers } from "hardhat";
+
+import { XVS, XVSStore, XVSVault } from "../../../typechain";
+
+const bigNumber18 = BigNumber.from("1000000000000000000"); // 1e18
+const rewardPerBlock = bigNumber18.mul(1);
+const lockPeriod = 300;
+const allocPoint = 100;
+const poolId = 0;
+
+interface XVSVaultFixture {
+  xvsVault: XVSVault;
+  xvs: XVS;
+  xvsStore: XVSStore;
+}
+
+describe("XVSVault", async () => {
+  let user1: Wallet;
+  let xvs: XVS;
+  let xvsVault: XVSVault;
+  let xvsStore: XVSStore;
+
+  before("get signers", async () => {
+    [user1] = await (ethers as any).getSigners();
```

Comment:
nit: we typically call the first signer the root/ deployer.  Could we use getNamedSigners here?

I haven't finished thinking this through but it seems like we could setup different named accounts to use for different scenarios tests

Reply:
I have updated the tests to use first account as `deployer` and second account for `user`